packages/service-provider-core/src/whatwg-url.ts

```diff
@@ -6,11 +6,21 @@ if (typeof require('util').TextDecoder !== 'function' || typeof require('util').
   Object.assign(require('util'), textEncodingPolyfill());
 }
 
-import { URL as WhatWGURL } from 'whatwg-url';
-import { URL as NodeURL } from 'url';
-export const URL = NodeURL ?? (WhatWGURL as typeof NodeURL);
+let URL: typeof import('url').URL;
 
-export function textEncodingPolyfill(): any {
+// Easiest way to get global `this` in any environment
+// eslint-disable-next-line no-new-func
+const globalThis = new Function('return this')();
+
+// Dynamically requiring URL should allow us to skip imports that potentially
+// can break browser runtimes
+if ('URL' in globalThis) {
+  URL = globalThis.URL;
+} else {
+  URL = require('url').URL ?? require('whatwg-url').URL;
```

Comment:
I have the suspect that `require('url')` is never reached though in node now.

Reply:
Not in current Node.js versions, yes

Reply:
Dropped this require in latest changes as we are supporting >=12 and URL should be globally available since >=10